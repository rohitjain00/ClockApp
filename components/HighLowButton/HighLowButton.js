import React from 'react';
import {StyleSheet, Text, View, Button} from 'react-native';
import PropTypes from 'prop-types';

export function HighLowButton(props) {
  function increase() {
    props.onChange(props.value + 2);
  }
  function decrease() {
    props.onChange(props.value - 2);
  }
  return (
    <View style={styles.container}>
      <Text>{props.text}</Text>
      <View style={styles.buttonContainer}>
        <Button
          title='&uarr;'
          style={styles.button}
          onPress={() => {
            increase();
          }}/>
      </View>
      <View style={styles.buttonContainer}>
        <Button
          title='&darr;'
          style={styles.button} onPress={() => {
            decrease();
          }}/>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  buttonContainer: {
    marginTop: 10,
  },
  container: {
    margin: 20,
  },
});

HighLowButton.propTypes = {
  onChange: PropTypes.func,
  text: PropTypes.string,
  value: PropTypes.value
}
