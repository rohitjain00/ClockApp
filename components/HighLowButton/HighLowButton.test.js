//CustomColorPicker.test.js
import React from 'react';
import {render} from '@testing-library/react-native';
import {HighLowButton} from "./HighLowButton";
import {configure, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe('HighLowButton Component', () => {
    const mockOnPress = jest.fn();
    const component = <HighLowButton onChange={mockOnPress} text={'testing component'} value={40}/>
    let wrapper;
    beforeEach(() => {
        wrapper = mount(component);
        jest.clearAllMocks();
    })
    it('High Low Button Props Validation', async () => {
        const componentProps = wrapper.props();
        expect(componentProps.onChange).toEqual(mockOnPress);
        expect(componentProps.text).toEqual('testing component');
        expect(componentProps.value).toEqual(40);
    });
    it('High Low Button Snapshot Validation', async () => {
        const tree = render(component).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('High Low Button Action Validation', async () => {
        // https://github.com/enzymejs/enzyme/issues/2263#issuecomment-568585047
        // According to above statement We cannot interact with state inside a function with useState.
    });
    it('High Low Button onChange handler', async () => {
        wrapper.props().onChange(38);
        expect(mockOnPress).toHaveBeenCalled();
        expect(mockOnPress).toHaveBeenCalledTimes(1);
        wrapper.props().onChange(42);
        expect(mockOnPress).toHaveBeenCalledTimes(2);
    });
});