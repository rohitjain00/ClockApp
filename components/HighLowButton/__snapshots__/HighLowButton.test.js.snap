// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HighLowButton Component High Low Button Snapshot Validation 1`] = `
<View
  style={
    Object {
      "margin": 20,
    }
  }
>
  <Text>
    testing component
  </Text>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          ↑
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          ↓
        </Text>
      </View>
    </View>
  </View>
</View>
`;
